### PR TITLE
	modified:   configuration/arm64/docker-compose.yml

### PR DESCRIPTION
**Pull request by jwillmartin**
modified:   configuration/arm64/initialization.sh
	modified:   configuration/arm64/mysql/add_v2xhub_user.bash

# PR Details
## Description

Updated arm64 configuration files to fix initial setup, which was no longer working. 

Updated: 
mariabd-client version
docker compose version
docker-compose.yml 

## Related Issue

#594 
## Motivation and Context

The initialization.sh and add_v2xhub_user.bash scripts were no longer working in arm64 configuration

## How Has This Been Tested?

Tested on latest Raspberry Pi OS on a Pi 4. Initialization and user addition worked. 

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[V2XHUB Contributing Guide](https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: configuration/arm64/docker-compose.yml [hunk 1]
```diff
@@ -1,5 +1,3 @@
-version: '3.7'
```
Comment (paulbourelly999) [inline configuration/arm64/docker-compose.yml:1]: @jwillmartin  Why are we removing this version

File: configuration/arm64/docker-compose.yml [hunk 2]
```diff
@@ -16,14 +14,15 @@ services:
       - mysql_root_password
     volumes:
       - ./mysql/localhost.sql:/docker-entrypoint-initdb.d/localhost.sql
-      - ./mysql/port_drayage.sql:/docker-entrypoint-initdb.d/port_drayage.sql
+      # - ./mysql/port_drayage.sql:/docker-entrypoint-initdb.d/port_drayage.sql
```
Comment (paulbourelly999) [inline configuration/arm64/docker-compose.yml:17]: Is the idea here that we do not want to default to having port drayage functionality on?

File: configuration/arm64/initialization.sh [hunk 3]
```diff
@@ -9,9 +9,6 @@ sudo apt update -y && sudo apt upgrade -y
 sudo apt-get install chromium-browser -y      #Chrome required for CARMA platform/V2X Hub UI(?)
 sudo apt install curl -y                      #Curl for downloading files over internet
 
-#install docker
-curl -L https://raw.githubusercontent.com/usdot-fhwa-stol/carma-platform/develop/engineering_tools/install-docker.sh | bash
```
Comment (paulbourelly999) [inline configuration/arm64/initialization.sh:13]: Why is this removed

Comment (paulbourelly999) [reply configuration/arm64/initialization.sh:13]: Why was this ever here.

Comment (paulbourelly999) [reply configuration/arm64/initialization.sh:13]: Can this be removed everywhere

File: configuration/arm64/mysql/add_v2xhub_user.bash [hunk 4]
```diff
@@ -7,7 +7,7 @@ arch=$(dpkg --print-architecture)
 if [ $arch = "amd64" ]; then
     REQUIRED_PKG="mysql-client"
 else
-    REQUIRED_PKG="mariadb-client-10.5"
+    REQUIRED_PKG="mariadb-client"
```
Comment (paulbourelly999) [inline configuration/arm64/mysql/add_v2xhub_user.bash:10]: What does removing version accomplish?

File: configuration/arm64/docker-compose.yml (hunk 2 above)
Comment (jwillmartin) [reply configuration/arm64/docker-compose.yml:17]: @paulbourelly999 Yeah. It's not used by anybody at the moment. It'll be better to have it commented out unless somebody will use it.

File: configuration/arm64/initialization.sh (hunk 3 above)
Comment (jwillmartin) [reply configuration/arm64/initialization.sh:13]: @paulbourelly999 I'm not sure why it was added, but Docker is installed below using the same steps in this script. I can remove it from the amd64 file, too.

File: configuration/arm64/mysql/add_v2xhub_user.bash (hunk 4 above)
Comment (jwillmartin) [reply configuration/arm64/mysql/add_v2xhub_user.bash:10]: @paulbourelly999 This version is not supported in the latest Raspberry Pi OS. Removing allows for the accepted version for the OS to be installed.

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
@jwillmartin Make the changes we talked about and then I am good to merge.

File: configuration/arm64/docker-compose.yml (hunk 2 above)
Comment (paulbourelly999) [reply configuration/arm64/docker-compose.yml:17]: Leave a comment indicating how to get port drayage functionality back and/or to come up with a more elegant solution to not including port drayage by default.

File: configuration/arm64/docker-compose.yml (hunk 1 above)
Comment (paulbourelly999) [reply configuration/arm64/docker-compose.yml:1]: Add this back

File: configuration/arm64/initialization.sh [hunk 5]
```diff
@@ -38,17 +35,21 @@ fi
...
-OS=$(lsb_release -i | awk 'FS=":" {print $3;}' | awk '{print tolower($0)}')
-arch=$(dpkg --print-architecture)
-curl -fsSL https://download.docker.com/linux/$OS/gpg | sudo apt-key add -
-sudo add-apt-repository "deb [arch=$arch] https://download.docker.com/linux/$OS $(lsb_release -cs) stable"
-sudo apt-get -y install docker-ce docker-ce-cli containerd.io docker-compose-plugin
-sudo apt -y install python3-pip
-sudo pip3 install docker-compose
-sudo apt update -y && sudo apt upgrade -y
-sudo docker-compose up -d
+for pkg in docker.io docker-doc docker-compose podman-docker containerd runc; do sudo apt-get remove $pkg; done
```
Comment (jwillmartin) [inline configuration/arm64/initialization.sh:38]: Was getting errors, updated and tested using Docker's official Debian installation steps

File: configuration/arm64/docker-compose.yml (hunk 2 above)
Comment (maefromm) [reply configuration/arm64/docker-compose.yml:20]: All, I think we need to leave the port_drayage as is. The expectation is that whatever was deployed in the past should be backwards compatible or scenarios would continue to work as is, unless we notify the client we're removing this feature or retiring it.

**Review by maefromm — CHANGES_REQUESTED** [review]
All, any changes that are done in arm64 that applies to amd64 should be done together and changes should be integration tested to make sure it works as expected without introducing any new issues.

File: configuration/arm64/docker-compose.yml (hunk 2 above)
Comment (jwillmartin) [reply configuration/arm64/docker-compose.yml]: Changed back

File: configuration/amd64/initialization.sh [hunk 6]
```diff
@@ -2,28 +2,24 @@
 directory=$(pwd)
 mysqlDir="$directory/mysql"
 
-# update and upgrade commands to update linux OS
+# Update and upgrade commands to update linux OS
 sudo apt update -y && sudo apt upgrade -y
 
-#installing necessary and useful apps
-sudo apt-get install chromium-browser -y      #Chrome required for CARMA platform/V2X Hub UI(?)
-sudo apt install curl -y                      #Curl for downloading files over internet
```
[the remark this answers is not in the document]
Comment (paulbourelly999) [reply configuration/amd64/initialization.sh:10]: This needs to be added back. It is still used

File: configuration/amd64/initialization.sh [hunk 7]
```diff
@@ -32,24 +28,30 @@ if test -f "$FILE2"; then
...
-sudo apt update -y && sudo apt upgrade -y
-sudo docker-compose up -d
-
-#create v2xhub user
+for pkg in docker.io docker-doc docker-compose podman-docker containerd runc; do sudo apt-get remove $pkg; done
+# Add Docker's official GPG key:
+sudo apt-get update
+sudo apt-get install ca-certificates curl
+sudo install -m 0755 -d /etc/apt/keyrings
+sudo curl -fsSL https://download.docker.com/linux/$(. /etc/os-release && echo "$ID")/gpg -o /etc/apt/keyrings/docker.asc
```
Comment (paulbourelly999) [inline configuration/amd64/initialization.sh:42]: Add curl back so this script does not fail for people without curl installed.

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
Only thing left is to add curl back.

**Review by paulbourelly999 — APPROVED** [review]
